server_addon/create_ayon_rez_addons.py

```diff
@@ -0,0 +1,205 @@
+import os
+import sys
+import re
+import json
+import shutil
+import argparse
+import zipfile
+import platform
+import collections
+from pathlib import Path
+from typing import Any, Optional, Iterable, Pattern, List, Tuple
+
+
+def read_addon_version(version_path: Path) -> str:
+    # Read version
+    version_content: dict[str, Any] = {}
+    with open(str(version_path), "r") as stream:
+        exec(stream.read(), version_content)
+    return version_content["__version__"]
+
+
+def create_addon_package(
+    addon_dir: Path,
+    output_dir: Path,
+    create_zip: bool,
+    keep_sources: bool
+):
+    addon_name = addon_dir.name
+    openpype_dir = None
+
+    if addon_name == "openpype":
+        openpype_dir = addon_dir / ".." / ".." / "openpype"
+        version_path = openpype_dir / "version.py"
+    else:
+        version_path = addon_dir / "server" / "version.py"
+
+    addon_version = read_addon_version(version_path)
+
+    addon_output_dir = output_dir / addon_dir.name / addon_version
+
+    if addon_output_dir.exists():
+        shutil.rmtree(str(addon_output_dir))
+
+    addon_output_dir.mkdir(parents=True)
+
+    shutil.copytree(
+        addon_dir,
+        addon_output_dir,
+        ignore=shutil.ignore_patterns(
+            '*.pyc',
+            '.*',
+            '*__pycache__*',
+        ),
+        dirs_exist_ok=True,
+    )
+
+    package_py = addon_output_dir / "package.py"
+
+    with open(package_py, 'w+') as pkg_py:
+        pkg_py.write(
+f"""name = "{addon_dir.name}"
+version = "{addon_version}"
+plugin_for = ["ayon_server"]
+build_command = "python {{root}}/rezbuild.py"
+"""
+        )
+
+    if addon_dir.name == "openpype":
+        ignored_hosts = []
+        ignored_modules = [
+            "ftrack",
+            "shotgrid",
+            "sync_server",
+            "example_addons",
+            "slack",
+            "kitsu",
+        ]
+        # Subdirs that won't be added to output zip file
+        ignored_subpaths = [
+            "*/addons/*",
+            "*/vendor/common/ayon_api/*",
+        ]
+        ignored_subpaths.extend(
+            f"*/hosts/{host_name}/*"
+            for host_name in ignored_hosts
+        )
+        ignored_subpaths.extend(
+            f"*/modules/{module_name}/*"
+            for module_name in ignored_modules
+        )
+
+        shutil.copytree(
+            openpype_dir,
+            addon_output_dir / "client",
+            ignore=shutil.ignore_patterns(*ignored_subpaths),
+            dirs_exist_ok=True,
+        )
+
+
+    if create_zip:
+        shutil.make_archive(
```

Comment:
`ZipFileLongPaths` was used because `shutil.make_archive` did crash on Windows when a certain path length of path was used.

Reply:
+ missing manifest file which is used when package is uploaded via frontend.

Reply:
The manifest is no longer needed in this instance, the info is in the `package.py` this will only work with the `rez` backend.

Reply:
> Even with enabling longer than 260 char paths in Windows?

Well, you have to enable it, and you have to know that you have to enable it, most people don't.